clients/src/main/java/org/apache/kafka/common/telemetry/ClientTelemetry.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.telemetry;
+
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Optional;
+
+/**
+ * A {@link MetricsReporter} may implement this interface to indicate support for collecting client
+ * telemetry on the client or server side.
+ */
+@InterfaceStability.Evolving
+public interface ClientTelemetry {
+
+    /**
+     * Implemented by the client {@link MetricsReporter} to provide a {@link ClientTelemetrySender}
+     * instance.
+     * <p>
+     * This instance may be cached by the client.
+     *
+     * @return Optional client side instance of {@link ClientTelemetrySender}.
+     */
+    default Optional<ClientTelemetrySender> clientSender() {
```

Comment:
I'm not familiar with ClientTelemetrySender.

Reply:
I have simplified ClientTelemetry.java in the updated PR.